lib/matplotlib/axes/_base.py

```diff
@@ -4142,22 +4150,26 @@ def get_tightbbox(self, renderer, call_axes_locator=True,
             self.apply_aspect()
 
         if self.axison:
-            bb_xaxis = self.xaxis.get_tightbbox(renderer)
+            igl = 'x' if for_layout_only else None
+            bb_xaxis = self.xaxis.get_tightbbox(renderer, ignore_label=igl)
             if bb_xaxis:
                 bb.append(bb_xaxis)
 
-            bb_yaxis = self.yaxis.get_tightbbox(renderer)
+            igl = 'y' if for_layout_only else None
+            bb_yaxis = self.yaxis.get_tightbbox(renderer, ignore_label=igl)
             if bb_yaxis:
                 bb.append(bb_yaxis)
-
         self._update_title_position(renderer)
-
         axbbox = self.get_window_extent(renderer)
         bb.append(axbbox)
 
         for title in [self.title, self._left_title, self._right_title]:
             if title.get_visible():
-                bb.append(title.get_window_extent(renderer))
+                bt = title.get_window_extent(renderer)
+                if for_layout_only and bt.width > 0:
+                    bt.x0 = (bt.x0 + bt.x1) / 2 - 0.001
+                    bt.x1 = bt.x0 + 0.002
```

Comment:
I got why you do this (get a nonzero but small width bbox so that the thing doesn't get cancelled out, centered at the right place) but it's a bit... mysterious and could perhaps use a comment.  Also, I think instead of picking an arbitraryish epsilon=0.001 here (and a different one for axis labels) you can use e.g. 0.5, as texts can't be less than one pixel wide anyways.